Microsoft.Packaging.Tools/tasks/Trimming/NuGetPackageNode.cs

```diff
@@ -12,15 +12,13 @@ namespace Microsoft.DotNet.Build.Tasks
     internal sealed class NuGetPackageNode : IIsIncluded
     {
         private const string NuGetPackageDependencies = "dependencies";
-        private List<NuGetPackageNode> _dependencies = new List<NuGetPackageNode>();
-        private IEnumerable<string> _dependencyIds;
+        private HashSet<NuGetPackageNode> _dependencies = new HashSet<NuGetPackageNode>();
```

Comment:
I actually don't want to have multiple copies of these things floating around.  I need to have an interconnected graph, not a bunch of parallel trees with "equal" objects that are different instances.  In that sense it is very much by-design that the caller maintains a collection of these for lookup: its not a cache.